### ISSUE TITLE
broken with LDC 1.16.0

### ISSUE THREAD

**Issue by WebFreak001** [skoppe/spasm#30]
I'm trying to run
```d
import spasm.spa;

@safe:
nothrow:

struct App
{
	mixin Node!"div";
	@prop innerText = "Hello World!";
}

mixin Spa!App;
```

```sdl
name "webapp"
dflags "-betterC"
dependency "spasm" version="~>0.2.0-beta.5"
```

but get

```
$ dub build --compiler=ldc2
Performing "debug" build using ldc2 for x86_64.
bolts 0.11.1: target for configuration "library" is up to date.
optional 0.16.0: target for configuration "unittest" is up to date.
mir-core 0.2.2: target for configuration "library" is up to date.
spasm 0.2.0-beta.5: target for configuration "library" is up to date.
webapp ~master: building configuration "application"...
Warning: Assuming critical section size = 40 bytes
ldc2 failed with exit code 1.
```

(also happens with `--build=release`)

```
LDC - the LLVM D compiler (1.16.0):
  based on DMD v2.086.1 and LLVM 8.0.0
  built with LDC - the LLVM D compiler (1.16.0)
  Default target: x86_64-unknown-linux-gnu
  Host CPU: znver1
```

and also wasm32 and wasm64 as registered targets.

**Comment by skoppe** [comment]
Yeah, that was a regression in ldc. You will have to add the `buildRequirements "allowWarnings"` line to dub.sdl. It is fixed in ldc 1.17.

Also see [BUILDING.md](https://github.com/skoppe/spasm/blob/master/BUILDING.md).

**Comment by WebFreak001** [comment]
thanks that fixed it, I thought the warning wouldn't be fatal because there was no file and line along with it
